dotCMS/src/main/java/com/dotcms/timemachine/business/TimeMachineAPIImpl.java

```diff
@@ -29,78 +29,85 @@
 public class TimeMachineAPIImpl implements TimeMachineAPI {
 
     @Override
-    public List<PublishStatus> startTimeMachine(List<Host> hosts, List<Language> langs, boolean incremental)  {
-        List<PublishStatus> list=new ArrayList<PublishStatus>(langs.size());
-        
-        
-        Date d = new Date();
-        
-        
-        
-        for(Language lang : langs) {
+    public List<PublishStatus> startTimeMachine(final List<Host> hosts,
+                                                final List<Language> languages,
+                                                final boolean incremental)  {
+
+        final List<PublishStatus> publishStatusList = new ArrayList<PublishStatus>(languages.size()); // todo: could it be immutable?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)